app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.java

```diff
@@ -0,0 +1,262 @@
+				noTrafficRestriction = getResources().getString(R.string.quest_maxheight_answer_noTrafficRestriction),
+				cantEstimate = getResources().getString(R.string.quest_maxheight_answer_cantEstimate);
+
+			final List<String> answers = new ArrayList<>(3);
+			answers.add(restrictsTraffic);
+			answers.add(noTrafficRestriction);
+			answers.add(cantEstimate);
+
+			DialogInterface.OnClickListener onSelect = new DialogInterface.OnClickListener()
+			{
+				Integer selection = null;
+
+				@Override public void onClick(DialogInterface dialog, int which)
+				{
+					if (which >= 0)
+					{
+						selection = which;
+						((AlertDialog)dialog).getButton(DialogInterface.BUTTON_POSITIVE).setEnabled(true);
+					}
+					else if (which == DialogInterface.BUTTON_POSITIVE)
+					{
+						if(selection == null || selection < 0 || selection >= answers.size()) return;
+						onAnswer();
+					}
+				}
+
+				private void onAnswer()
+				{
+					String answer = answers.get(selection);
+					Bundle data = new Bundle();
+					int type = 0;
+					if(answer.equals(restrictsTraffic))	type = IS_BELOW_DEFAULT;
+					if(answer.equals(noTrafficRestriction))	type = IS_DEFAULT;
+					if(answer.equals(cantEstimate))    type = IS_NOT_INDICATED;
+					data.putInt(NO_SIGN, type);
+					applyImmediateAnswer(data);
+				}
+			};
+
+			AlertDialog dlg = new AlertDialogBuilder(getActivity())
+				.setSingleChoiceItems(answers.toArray(new String[0]), -1, onSelect)
+				.setTitle(R.string.quest_maxheight_answer_noSign_question)
+				.setPositiveButton(android.R.string.ok, onSelect)
+				.setNegativeButton(android.R.string.cancel, null)
+				.show();
+
+			dlg.getButton(DialogInterface.BUTTON_POSITIVE).setEnabled(false);
+		});
+	}
+
+	@Override protected void onClickOk()
+	{
+		if(!hasChanges())
+		{
+			Toast.makeText(getActivity(), R.string.no_changes, Toast.LENGTH_SHORT).show();
+			return;
+		}
+
+		if(userSelectedUnrealisticHeight())
+		{
+			confirmUnusualInput(this::applyMaxHeightFormAnswer);
+		}
+		else
+		{
+			applyMaxHeightFormAnswer();
+		}
+	}
+
+	private boolean userSelectedUnrealisticHeight()
+	{
+		double height = getHeightFromInput().toDouble();
+		String heightUnit = getHeightFromInput().getUnit();
+
+		double heightInMeter = heightUnit.equals(Height.METRIC) ? height : feetToMeter(height);
+		return heightInMeter > 6 || heightInMeter < 2;
+	}
+
+	private static double feetToMeter(double feet)
+	{
+		return feet / 3.2808;
+	}
+
+	private void applyMaxHeightFormAnswer()
+	{
+		String height = getHeightFromInput().toString();
+
+		Bundle answer = new Bundle();
+		answer.putString(MAX_HEIGHT, height);
+		applyFormAnswer(answer);
+	}
+
+	private Height getHeightFromInput()
+	{
+		boolean isMetric = heightUnitSelect.getSelectedItem().equals("m");
+
+		if (isMetric)
+		{
+			String input = heightInput.getText().toString();
+			if (!input.isEmpty())
+			{
+				if (input.contains("."))
```

Comment:
Wouldn't that be a comma for German language (and whatnot for others)?

You should rather parse the string into a number (using the user's locale?) and then split up the number, if necessary. I don't really get why you do this for metric heights as well though. Can't you just have two constructors `Height(float meters)` and `Height(int feet, int inch)`?